beginner_source/aws_distributed_training_tutorial.py

```
@@ -0,0 +1,675 @@
+# dataset. To accomodate for the dimensionality mismatch of STL-10 with
+# Resnet18, we will resize each image to 224x224 with a transform. Notice,
+# the choice of model and dataset are orthogonal to the distributed
+# training code, you may use any dataset and model you wish and the
+# process is the same. Lets get started by first handling the imports and
+# talking about some helper functions. Then we will define the train and
+# test functions, which have been largely taken from the ImageNet Example.
+# At the end, we will build the main part of the code which handles the
+# distributed training setup. And finally, we will discuss how to actually
+# run the code.
+# 
+
+
+######################################################################
+# Imports
+# ~~~~~~~
+# 
+# The important distributed training specific imports here are
+# `torch.nn.parallel <https://pytorch.org/docs/stable/nn.html#torch.nn.parallel.DistributedDataParallel>`__,
+# `torch.distributed <https://pytorch.org/docs/stable/distributed.html>`__,
+# `torch.utils.data.distributed <https://pytorch.org/docs/stable/data.html#torch.utils.data.distributed.DistributedSampler>`__,
+# and
+# `torch.multiprocessing <https://pytorch.org/docs/stable/multiprocessing.html>`__.
+# It is also important to set the multiprocessing start method to *spawn*,
+# as the default is *fork* which may cause deadlocks when using multiple
+# worker threads for dataloading.
+# 
+
+import time
+import sys
+import torch
+import torch.nn as nn
+import torch.nn.parallel
+import torch.distributed as dist
+import torch.optim
+import torch.utils.data
+import torch.utils.data.distributed
+import torchvision.transforms as transforms
+import torchvision.datasets as datasets
+import torchvision.models as models
+from torch.multiprocessing import Pool, Process, set_start_method
+try:
+    set_start_method('spawn')
+except RuntimeError:
+    pass
```

Please raise the error here instead of swallowing it.